Documentation/compiling-seissol.rst

```diff
@@ -1,44 +1,96 @@
+used during all calculations. If your simulation gives an error when using single precision, it can sometimes be fixed by re-compiling for double precision.
+The ``ORDER`` parameter may range from 2 to 8. The ``EQUATIONS`` parameter may be ``elastic``, ``viscoelastic2``, ``poroelastic``,
+or ``anisotropic``. For the ``viscoelastic2`` setting, you will need to add e.g. ``-DNUMBER_OF_MECHANISMS=3`` (or replace 3 by any other number greater than zero).
+
+Once you have installed a GEMM generator, choose a different option in ``GEMM_TOOLS_LIST`` to obtain a faster version of SeisSol.
+For example, ``GEMM_TOOLS_LIST=libxsmm`` makes libxsmm generate the kernels.
+Also, you may set ``GRAPH_PARTITIONING_LIBS`` to a graph partitioning library: we currently allow the values
+``parmetis``, ``parhip`` and ``ptscotch``.
+
+The ``HOST_ARCH`` parameter tells the compiler for which CPUs to optimize for. Note in particular that using the wrong host architecture may lead to "illegal instruction" (SIGILL) errors.
+The host architecture ``hsw`` (an abbreviation for Haswell) will generate at most AVX2 instructions and should usually work when installing SeisSol on a local Linux machine for testing purposes. On clusters or workstations using Intel Skylake CPUs or newer (e.g. SuperMUC), use `skx` instead.
+For an overview, the following values are accepted by ``HOST_ARCH``:
+
+.. list-table:: Title
+   :widths: 20 40 40
+   :header-rows: 1
+
+   * - ``HOST_ARCH``
+     - Architecture
+     - Notes
+   * - ``noarch``
+     - No architecture-specific optimizations
+     - Generates plain x86-64 instructions, without SIMD instructions like SSE/AVX/AMX etc.
+   * - ``wsm``
+     - Intel Westmere architecture
+     - Generates SSE instructions (up to SSE 3).
+   * - ``snb``
+     - Intel Sandy Bridge architecture
+     - Generates AVX instructions.
+   * - ``hsw``
+     - Intel Haswell
+     - Generates AVX2 instructions.
+   * - ``skx``
+     - Intel Skylake-X (including Skylake-SP)
+     - Generates AVX-512{F,CD,BW,DQ,VL} instructions. (NOTE: Skylake desktop processors are NOT included here, unless they contain an "X" in their name, such as e.g. i9 7800X)
+   * - ``knc``
+     - Intel Knight's Corner (Xeon Phi coprocessor)
+     - Generates Knight's Corner-specific instructions.
+   * - ``knl``
+     - Intel Knight's Landing (Xeon Phi, optionally as coprocessor)
+     - Generates AVX-512{F,CD,PF,ER} instructions.
+   * - ``rome``
+     - AMD Rome (Zen, 2nd generation)
+     - Generates AVX2 instructions. For the libxsmm kernel generator, it is deemed equivalent to `hsw`.
+   * - ``thunderx2t99``
+     - ARM ThunderX2
+     - 
+   * - ``power9``
+     - IBM PowerPC 9
+     - 
+
+In case of a manual installation of dependencies (and not using the `setup.sh` script from the "Installing Dependencies" page), you may have to prepend :code:`CMAKE_PREFIX_PATH` and :code:`PKG_CONFIG_PATH` to the cmake command, e.g. for dependencies installed in :code:`${PREFIX}`:
```

Comment:
It is important that all dependencies and SeisSol are compiled with the same compiler. In particular, `gcc` and `intel` compilers are not compatible.
On some clusters, where several compilers are installed, you might want to specifically point to the compiler, which you want to use for SeisSol. For the `intel` compiler use e.g. `CC=mpiicc CXX=mpiicpc cmake ...`.

Reply:
gcc/clang however are *typically* not a problem

Reply:
Maybe: `In particular, we have experienced problems when mixing gcc/clang with intel compilers`.

Reply:
We don't particularly need to use `mpicc`/`mpiicc` (and `mpicxx`/`mpiicpc`), do we? I.e. the non-MPI compilers should be good... Or are they?

Reply:
In my experience, using the non-mpi compilers, typically cmake can't find the proper libraries. That's why I always use the mpi wrappers.

Reply:
I've tested the non-mpi compilers (on a gcc12 openmpi compilation) and it was working fine.